### ISSUE TITLE
Alot of errors when running game built

### ISSUE THREAD

**Issue by AndersMalmgren** [microsoft/ProjectAcoustics#33]
The log is spammed with these

(Filename: C:\buildslave\unity\build\Runtime/Scripting/ManagedReference/SerializableManagedRef.cpp Line: 199)

The referenced script (Microsoft.Cloud.Acoustics.AcousticsGeometry) on this Behaviour is missing!

**Comment by kegodin** [comment]
Hi Anders, thanks for filing this issue. Could you please provide some more repro steps or information about context? For example:
* Is this a blocking problem, or are these warnings/errors something you've worked around?
* What operation was running when these log entries were written?
* Is this for a bake or during runtime? If bake, is this for a local bake or a cloud bake?

**Comment by AndersMalmgren** [comment]
hi Keith. It seems there is a bug in your AcousticsGeometry tagging workflow. You tag all meshes that you want to include in the bake with the AcousticsGeometry component. Then when you compile the game and run it standalone yuo get all these errors when loading a scene. 


Its not completely blocking, but it fills up the log with thousands of these errors, thats not nice, plus its probably a bit degrading performance wise when loading scenes.

**Comment by kegodin** [comment]
Great for the info, understood, it arises when building standalone. One last detail: could you share the exact Unity version? We'll look into this. Thanks.

**Comment by kegodin** [comment]
For our internal tracking I'll note the ADO bug # 25599318.

**Comment by AndersMalmgren** [comment]
Sure its 2019.2.21 but probably same on 2019.3.5

**Comment by kegodin** [comment]
Thanks for the info. We still do a fair amount of testing on Unity's LTS version (2018.4.latest) so it's helpful to have it narrowed down to 2019.2.

**Comment by AndersMalmgren** [comment]
@kegodin I just tried in 2019.3.5 in a empty project, imported the project acoustics project set the demo scene to loading scene, built and ran the game. Same errors in the log

**Comment by AndersMalmgren** [comment]
@MikeChemi Hi, when will the fix be available? Thanks

**Comment by MikeChemi** [comment]
Hi @AndersMalmgren Good to hear from you. I don't currently have an estimate for when we'll be releasing an update, but this particular issue is easy to fix. Open up AcousticsGeometry.cs and reomve the "#if UNITY_EDITOR" and corresponding "#endif" at the end of the file. Should be good to go after that.

**Comment by AndersMalmgren** [comment]
@MikeChemi Thanks mike you too!

Ah, yeah, that workaround have already been done by us. But I though the fix was on the other side of things since that component is only needed to mark the geo when baking scene. Not a biggy but keeping it will increase memory etc.

**Comment by AndersMalmgren** [comment]
I think there is a build event that you can subscribe to and remove those components. Will have a look at report back

**Comment by AndersMalmgren** [comment]
Hi again @MikeChemi this seem to work

```csharp
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.


using UnityEditor;
using UnityEngine;

#if UNITY_EDITOR && !NET_4_6
#error Microsoft Project Acoustics plugin requires "Scripting Runtime Version" and "API Compatibility Level" under Edit/Project Settings/Player set to ".NET 4.x". Please change the settings and restart the editor.
#endif

#if UNITY_EDITOR
using UnityEditor.Callbacks;

namespace Microsoft.Cloud.Acoustics
{
    [DisallowMultipleComponent]
    public class AcousticsGeometry : MonoBehaviour
    {
        // This component has no content. Its presence on a GameObject indicates
        // that the object should be used in cloud acoustics calculations.

        [PostProcessScene]
        public static void DestroyOnBuild()
        {
            if (BuildPipeline.isBuildingPlayer)
            {
                foreach (var geo in FindObjectsOfType<AcousticsGeometry>())
                {
                    DestroyImmediate(geo);
                }
            }
        }
    }
}
#endif
```

It will not remove the component while in editor. It will only remove them from the build.

**Comment by MikeChemi** [comment]
This is awesome! Thanks, @AndersMalmgren !!

**Comment by AndersMalmgren** [comment]
> This is awesome! Thanks, @AndersMalmgren !!

Glad to be of service :)

**Comment by AndersMalmgren** [comment]
@MikeChemi Hi again. Above code will only remove component from scenes. It will still be present on prefabs. Probably not a biggy since Acoustics only support static objects ad designtime. But if you intance a prefab with the geo component you wll get a missing error like above. Also I guess down the road you want to support dynamic occlusion like doors.

In 2020.x this mechanic was added that let you preprocess assets

https://docs.unity3d.com/2020.2/Documentation/ScriptReference/AssetPostprocessor.OnPostprocessPrefab.html

**Comment by MikeChemi** [comment]
We included this fix in 2.0 #55
